microprofile/WFLY-12313_MicroProfile_OpenAPI_1.1.adoc

```diff
@@ -0,0 +1,131 @@
+* Support YAML and JSON for both input (static/pre-generated OpenAPI document) and output formats.
+* MicroProfile OpenAPI TCK must pass
+
+=== Nice-to-Have Requirements
+
+* Optional features of the specification:
+** Multiple deployments:
+*** If a container allows multiple deployments, the specification expects the `/openapi` endpoint to return the logical union of the documented services for all applications in the runtime. See: https://download.eclipse.org/microprofile/microprofile-open-api-1.1.2/microprofile-openapi-spec.html#_multiple_applications
+**** Initially, we will limit multiple deployment support to a single deployment per host (for a given endpoint). i.e. each deployment, for which the OpenAPI endpoint is enabled, must either use a distinct host, or a distinct endpoint path (if a given virtual host is associated with multiple deployments).
+**** Merging (and more interestingly, unmerging on undeploy) OpenAPI models for all applications deployed to a given host.
+***** Consider for a future enhancement
+** HTTPS transport for the `/openapi` endpoint.
+*** If the server/host of a given application defines an HTTPS listener, we will allow it, but not require it.
+*** Additionally, if the Undertow server/host associated with the application does not define an HTTP endpoint (as required by the spec), we should log a WARN message.
+** The MicroProfile OpenAPI specification suggests, but does not require, the use of a `format=JSON` or `format=YAML` parameter for the `/openapi` endpoint, used to define content type of the resulting document, in lieu of an `Accept` header.
+*** This is very reasonable, and is simple enough to do.  See: https://download.eclipse.org/microprofile/microprofile-open-api-1.1.2/microprofile-openapi-spec.html#_query_parameters
+* Use a `mp.openapi.extensions.enabled` property to disable OpenAPI documentation globally (via mp-config subsystem) or per-application (e.g. /META-INF/mp-config.properties).
+** Provides a parameterizable means of disabling OpenAPI documentation in specific environments (e.g. prod vs dev)
+** Provides a mechanism to control which application associated with a given virtual host should generate an MicroProfile OpenAPI model.
+* Use a `mp.openapi.extensions.path` property to customize the path of the OpenAPI endpoint.
+** Provides a means for generating OpenAPI documentation for multiple applications associated with the same virtual host.
+** Potentially useful to support legacy OpenAPI tooling (e.g. `/swagger`)
+** Analogous to the `quarkus.smallrye-openapi.path` property in Quarkus.
+* Use a `mp.openapi.extensions.relative` property to configure whether auto-generated Server records should be absolute or relative.
+** When relative, the generated record will contain the context path of the deployment
+** When absolute, generated records will include all hosts/ports at which the given deployment is accessible.
```

Comment:
I think it would be good to have the allowed values for the property here, or just to mention that it is expecting a Boolean. At first sight it looks like an enumeration description.

Reply:
Sure.